### PR TITLE
Use preset-env

### PR DESCRIPTION
**Pull request by hzoo**
> https://github.com/babel/babel-preset-env

Instead of preset-es2015.

I would of targeted node 6.1.0 but it's bundling with webpack and minifying with uglify atm.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,4 +1,10 @@
 {
-  "presets": ["es2015", "react"],
-  "plugins": ["transform-class-properties", "transform-object-rest-spread"]
+  "presets": [
+    "env",
```
Comment (hzoo) [inline .babelrc]: Using uglify but otherwise this could be

```js
    ["env", {
      "targets": {
        "node": 6.1
      }
    }],
```

Comment (sebmarkbage) [inline .babelrc]: Note that in this PR I also use some transforms directly as a hack while we're still working out native support for them: https://github.com/facebook/prepack/pull/397/files#r114634310

Comment (hzoo) [inline .babelrc]: Ah ok. Well without any targets it acts the same as preset-es2015 other than also providing support for es2016/es2017. env includes the same dependencies in package.json

Comment (hzoo) [inline .babelrc]: Yeah `env` includes all the transforms in e2015 (superset) so it's ok

**Comment by sebmarkbage** [comment]
The reason we might not want this is if we want to explicit limit ourselves to Node.js compatible syntax so that we can avoid compiling at all in the future. But we could probably just change the few places where that wouldn't work if that happens.

**Comment by hzoo** [comment]
Yeah, if you want you can change the value of the node version to compile less. In development you can also do `"node": "current"` to run on the currently running version

```js
{
  "presets": [
    ["env", {
      "targets": {
        "node": "7"
      }
    }]
  ]
}

**Comment by hzoo** [comment]
FYI, Also I noticed this PR added a yarn.lock file and it wasn't there before so don't know if that was wanted or not.

**Pull request merged**